components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaFetchRecords.java

```diff
@@ -516,14 +543,16 @@ public BridgeExceptionHandlerToErrorHandler getBridge() {
      * or via JMX
      */
     public void pause() {
-        consumer.pause(consumer.assignment());
+        LOG.info("A pause request was issued and the consumer thread will pause after current processing has finished");
+        state = State.PAUSE_REQUESTED;
```

Comment:
Maybe we should only allow to pause if the state is `RUNNING `?

Reply:
Actually, I think it should be either `RUNNING` or `RESUME_REQUESTED` (or, basically !PAUSED) . There's always a chance that it could receive a pause request almost immediately after a pause one.